src/core/Engine.ts

```diff
@@ -1,106 +1,109 @@
-import * as PIXI from 'pixi.js'
 import { System, Entity } from 'src/core'
-import { ShapeComponent, Transform, BoxShape } from 'src/components'
-import { BOX_SHAPE, ComponentType } from 'src/components/types'
-import { uuid } from 'src/utils'
+import { EventManager, EventCallback } from './EventManager'
 
+/**
+ * The Engine keeps track of all ECS-related entities and systems.
+ * It a singleton and should only be lazily instanced via the GetInstance() method.
+ * @fires entity_added
+ * @fires entity_removed
+ * @fires system_added
+ * @fires system_removed
+ */
 export class Engine {
   systems: Record<string, System> = {}
   entities: Record<string, Entity> = {}
-  engineObjects: Record<string, PIXI.Container> = {}
-  app: PIXI.Application
 
-  private static instance: Engine
+  private static Instance: Engine
+  private eventManager: EventManager = new EventManager()
+  private pool: Entity[] = []
 
-  private componentHandlers = {
-    transform: this.handleTransform,
-    shape: this.handleShape
-  }
-
-  static getInstance(app: PIXI.Application): Engine {
-    if (!Engine.instance) {
-      Engine.instance = new Engine(app)
+  static GetInstance(): Engine {
```

Comment:
Why are we changing `getInstance` to `GetInstance`? I know it's just convention but it's kind of weird to have a method starting with a capital letter

Reply:
Just a convention for static methods. When I copied Vector2 it came with some uppercase static methods and I know other languages use the convention as well /shrug

Should we change it?

Reply:
I'm not really used to static methods being uppercased, but I'm speaking from a front-end/old-fart js point of view. 
I don't really mind if we keep it consistent but I think it's worth considering if we ever decide to abstract the ECS part away, as other projects might be using this convention or the more js-based one

Reply:
let's refactor it in the future if needed, some libs do it lowercase some do uppercase

tabs vs spaces :P